[VA-2888] Update XDM schema to include all Media Analytics properties

midomir opened the pull request:
Issue #700

lrosenthol commented on schemas/context/media-timed-asset-reference.schema.json:
```diff
@@ -21,11 +21,39 @@
           "format": "uri-reference",
           "description": "Identifier of the content, which can be used to tie back to other industry or CMS IDs."
         },
+        "dc:creator": {
```
Shouldn't these be references to the master DC grammar and not copied here?

midomir replied:
Changed this with a reference to DC grammar.

lrosenthol commented on schemas/context/media-timed-asset-reference.schema.json:
```diff
@@ -21,11 +21,39 @@
+          },
+          "description": "An entity primarily responsible for making the resource."
+        },
+        "dc:publisher": {
```
Shouldn't these be references to the master DC grammar and not copied here?

midomir replied:
Changed this with a reference to DC grammar.

lrosenthol commented on schemas/context/media-timed-asset-reference.schema.json:
```diff
@@ -21,11 +21,39 @@
           "type": "string",
           "description": "The friendly, human-readable name of the timed media asset."
         },
+        "xdm:recordLabel": {
```
Isn't there a pre-existing media grammar that could be used?  Do we really need to invent a new one??

midomir replied:
Changed this with a reference to ID3.

lrosenthol commented on schemas/context/media-timed-asset-reference.schema.json:
```diff
@@ -21,11 +21,39 @@
+          "type": "string",
+          "description": "The name of the record label associated with the music recording or video."
+        },
+        "xdm:artist": {
```
Isn't there a pre-existing media grammar that could be used?  Do we really need to invent a new one??

midomir replied:
Changed this with a reference to XMP grammar.

lrosenthol commented on schemas/context/media-timed-asset-reference.schema.json:
```diff
@@ -21,11 +21,39 @@
+          "type": "string",
+          "description": "The name of the album artist or group performing the music recording or video."
+        },
+        "xdm:album": {
```
Isn't there a pre-existing media grammar that could be used?  Do we really need to invent a new one??

midomir replied:
Changed this with a reference to XMP grammar.

lrosenthol commented on schemas/context/media-timed-asset-view-details.schema.json:
```diff
@@ -26,6 +26,16 @@
           "type": "string",
           "description": "Name of the content player."
         },
+        "xdm:videoSegment": {
```
Isn't there a pre-existing media grammar that could be used?  Do we really need to invent a new one??

midomir replied:
This is something that we defined and computed in Video Analytics to describe what part of the content was viewed. It is computed in terms of the min and max of the playhead values during a playback session. As such, it doesn't mean anything generic that could be found in a pre-existing grammar.

lrosenthol commented on schemas/context/media-timed-asset-view-details.schema.json:
```diff
@@ -26,6 +26,16 @@
+          "type": "string",
+          "description": "The interval that describes the part of the content that has been viewed in minutes."
+        },
+        "xdm:path": {
```
Isn't there a pre-existing media grammar that could be used?  Do we really need to invent a new one??

midomir replied:
This has the same meaning as the "xdm:path" field in schemas/assets/image.schema.json and schemas/assets/video.schema.json. We used the same name for consistency.

lrosenthol commented on schemas/context/media-timed-asset-view-details.schema.json:
```diff
@@ -55,6 +65,11 @@
           "type": "string",
           "description": "The programmer network name."
         },
+        "xdm:radioStation": {
```
Isn't there a pre-existing media grammar that could be used?  Do we really need to invent a new one??

midomir replied:
Changed this with a reference to ID3.

lrosenthol commented on schemas/context/media-timed-asset-view-details.schema.json:
```diff
@@ -74,10 +89,15 @@
           "$ref": "https://ns.adobe.com/xdm/context/media-timed-qoe",
           "description": "Timed media properties related to the content playback's quality of experience (QoE)."
+        },
+        "xdm:downloadedPlayback": {
```
Isn't there a pre-existing media grammar that could be used?  Do we really need to invent a new one??

midomir replied:
This is something that we defined and computed in Video Analytics to describe what type of content was played. As such, it doesn't mean anything generic that could be found in a pre-existing grammar.

lrosenthol commented on schemas/data/ad-count.schema.json:
```diff
@@ -0,0 +1,56 @@
+      "properties": {
+        "@id": {
+          "type": "string",
+          "const": "xdm:adCount",
```
This CANNOT be the valid for `@id`

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/data/ad-count.schema.json:
```diff
@@ -0,0 +1,56 @@
+          "const": "xdm:adCount",
+          "description": "The unique identifier of this metric."
+        },
+        "schema:name": {
```
That doesn't seem like the right name - copy/paste error?

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/data/ad-count.schema.json:
```diff
@@ -0,0 +1,56 @@
+          "description": "How to take measures of this metric.",
+          "const": "count"
+        },
+        "xdm:unit": {
```
Clearly an incomplete editing error

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/data/ad-count.schema.json:
```diff
@@ -0,0 +1,56 @@
+        "xdm:measurement": {
+          "type": "string",
+          "description": "How to take measures of this metric.",
+          "const": "count"
```
????

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/data/average-minute-audience.schema.json:
```diff
@@ -0,0 +1,56 @@
+      "properties": {
+        "@id": {
+          "type": "string",
+          "const": "xdm:averageMinuteAudience",
```
???

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/data/average-minute-audience.schema.json:
```diff
@@ -0,0 +1,56 @@
+          "const": "xdm:averageMinuteAudience",
+          "description": "The unique identifier of this metric."
+        },
+        "schema:name": {
```
Again, incorrect key here

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/data/average-minute-audience.schema.json:
```diff
@@ -0,0 +1,56 @@
+          "const": "metric: averageMinuteAudience",
+          "description": "Describes the average content time spent for a specific media item - i.e. the total content time spent divided by the length for all of the playback sessions."
+        },
+        "xdm:measurement": {
```
looks like more copy/paste and then incomplete editing

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/data/chapter-count.schema.json:
```diff
@@ -0,0 +1,56 @@
+  ],
+  "description": "Describes the number of chapters started during the media session.",
+  "definitions": {
+    "metricdefinition": {
```
Another set of copy/paste and incomplete editing...

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/data/unique-time-played.schema.json:
```diff
@@ -0,0 +1,56 @@
+  ],
+  "description": "Describes the sum of the unique intervals seen by a user on a timed media asset - i.e. the length playback intervals viewed multiple times are only counted once.",
+  "definitions": {
+    "metricdefinition": {
```
And another...

midomir replied:
Taken out this file as it was using the legacy metricdefinition. Using measure instead.

lrosenthol commented on schemas/context/experienceevent-media.example.1.json:
```diff
@@ -123,7 +148,9 @@
-      }
+      },
+      "xdm:idp": "Hulu",
+      "xdm:mediaAuth": "FALSE"
```
improper encoding of a boolean value

midomir replied:
This is not a boolean variable. If usually has a true/false value, but the clients can send any text value. Will add some extra examples with different values to make it more clear.

lrosenthol commented on schemas/context/experienceevent.example.4.json:
```diff
@@ -143,7 +168,9 @@
-      }
+      },
+      "xdm:idp": "MTV",
+      "xdm:mediaAuth": "true"
```
again, bad boolean

midomir replied:
This is not a boolean variable. If usually has a true/false value, but the clients can send any text value. Will add some extra examples with different values to make it more clear.

cmathis commented on schemas/common/extensible.schema.json:
```diff
@@ -225,6 +225,11 @@
+                  "type": "string",
+                  "format": "url",
+                  "const": "http://id3.org/ns/2.4/"
                 }
```
In addition, we need to add id3 to the context.jsonld file here - https://github.com/adobe/xdm/blob/master/schemas/common/context.jsonld

midomir replied:
Done

cmathis commented on schemas/context/media-timed-asset-reference.schema.json:
```diff
@@ -26,6 +26,11 @@
+        "id3:TPUB": {
+          "title": "Record Label",
+          "description": "The name of the record label associated with the music recording or video.",
+          "$ref": "http://id3.org/ns/2.4/TPUB"
```
The $ref is intended to point to a schema in this github repo with $id=http://id3.org/ns/2.4/TPUB. You cannot reference an external URL or something does that not exist.

lrosenthol replied:
Was just talking about this one this morning @cmathis .   Why can't it be used to refer to an external namespace?

midomir replied:
Fixed this

lrosenthol commented on schemas/context/advertising.example.1.json:
```diff
@@ -18,6 +18,13 @@
     }
   },
+  "xdm:adAssetReference": {
+    "@id": "https://data.adobe.io/entities/ad-id/AD1",
```
Is this really an AdID (https://www.ad-id.org/)?  If so, we need a reference to it.  If not, then someone needs to give it a new name...

midomir replied:
Referenced AdID in the description.

lrosenthol commented on schemas/context/media-timed.schema.json:
```diff
@@ -109,6 +129,16 @@
           "description": "Describes the duration in seconds in which playback was paused by the user."
+        },
+        "xdm:idp": {
+          "title": "MVPD Identifier",
```
Can you spell out MVPD and/or a link to the spec?

midomir replied:
Added a link to the spec and spelled it out in the description.

lrosenthol commented on schemas/context/media-timed.schema.json:
```diff
@@ -109,6 +129,16 @@
+        "xdm:idp": {
+          "title": "MVPD Identifier",
+          "type": "string",
+          "description": "The MVPD provided via Adobe authentication."
```
So this is an internal Adobe field??

midomir replied:
No, corrected the description.

lrosenthol commented on schemas/context/media-timed.schema.json:
```diff
@@ -109,6 +129,16 @@
+        "xdm:mediaAuth": {
+          "title": "Media authorized",
+          "type": "string",
+          "description": "The user has been authorized via Adobe authentication."
```
This is only good for Adobe authentication??

midomir replied:
Yes, this is specific to our product and to Adobe authentication. The accepted values are not only true/false as they would be for a generic "Media authorized" field, it could also contain client custom data.

lrosenthol replied:
Hmm...this is a problem since Adobe-specific stuff can't be part of a public schema.  Can you move these values to a separate schema, in the Adobe section, and then mix them in?

midomir replied:
Checked this again and it seems it's Adobe authentication for our product, but it's actually not Adobe specific. Corrected the description.

lrosenthol commented on schemas/context/media-timed-asset-view-details.schema.json:
```diff
@@ -26,6 +26,16 @@
+        "xdm:videoSegment": {
+          "title": "Video segment",
+          "type": "string",
+          "description": "The interval that describes the part of the content that has been viewed in minutes."
```
Could we at least tie this to a standard, such as Media Fragments (https://www.w3.org/TR/media-frags/#naming-time)?

midomir replied:
This doesn't follow the Media Fragments standard, it's something of the format 
[start minute]-[end minute], where both bounds of the interval are compulsory. Added a regex pattern to the field to specify this.

midomir commented:
Added some additional changes from some new requirements from @scrisan to this PR as they referred to some of the new fields we are adding. We changed the type of some fields from number to integer and we changed the title of some fields to ensure it is aligned with the field name in the reports.

stormybriggs1 commented on schemas/context/media-timed.schema.json:
```diff
@@ -31,84 +31,133 @@
         },
         "xdm:impressions": {
-          "title": "Impressions",
-          "$ref": "https://ns.adobe.com/xdm/data/measure",
```
Is this a breaking change to remove `$ref`? If so I have concerns about this change.

stormybriggs1 replied:
I have confirmed this is a breaking change. I would prefer that we only add to the schema rather than break backwards compatibility.

midomir replied:
Reverted the changes for removing the references to measure so we can separate the two issues we were working on. This PR now only contains the addition of new fields.

stormybriggs1 replied:
Thank you for making this additive only! I have no further concerns.

midomir commented:
Reverted the changes for removing the references to measure so we can separate the two issues we were working on. This PR now only contains the addition of new fields.
